### DISCUSSION TITLE
Passing props to InteractionsProvider.attach()

### DISCUSSION THREAD

**Discussion by dhrblav** [molstar/molstar#556] [Q&A]
Sorry for this maybe trivial question.
I am trying to call `InteractionsProvider.attach(ctx: CustomProperty.Context, data: Structure, props: Partial<InteractionsProps>)` with props to include ionic interactions.
So I write this:
`await interactionsProvider.attach(ctx, structure,
{
    providers: {
        ionic: {
            name: 'on',
            params: {
                distanceMax: PD.Numeric(5.0, { min: 0, max: 8, step: 0.1 })
            }
        }
    }
}
);`

But this gives a type error:
Type '{ ionic: { name: "on"; params: { distanceMax: PD.Numeric; }; }; }' is not assignable to type 'Normalize<{ ionic: NamedParams<{}, "off"> | NamedParams<{ distanceMax: Numeric; }, "on">; 'pi-stacking': NamedParams<{}, "off"> | NamedParams<{ ...; }, "on">; ... 5 more ...; 'metal-coordination': NamedParams<...> | NamedParams<...>; }>'

Can you please help with this?

**Comment by dsehnal** [comment]
Check ``updateInterationProps`` in ``src\mol-plugin-state\manager\structure\component.ts``
